### DISCUSSION TITLE
Hiding fields on create

### DISCUSSION THREAD

**Discussion by tonyellow** [directus/directus#16370] [Archive]
I am currently assessing Directus and looking for a way to hide form elements on create. A solution I found is to add a condition to the field which hides the element when the id is null:

![Screenshot 2022-11-09 at 10 44 27](https://user-images.githubusercontent.com/8984917/200796748-0c61ac16-b54f-4508-a0d2-a63d8c9d6c6d.png)

This was working untill i started structuring my fields into "Detail groups":

![Screenshot 2022-11-09 at 10 47 59](https://user-images.githubusercontent.com/8984917/200797390-2e789c03-009f-4e39-83b7-9a9f1b2518f5.png)

Why is the "id is null check" not working anymore after grouping my fields? I think it's a bug and I need to report it. It there a better way to hide fields on create, eg through an extension?

**Comment by rijkvanzanten** [comment]
>  It there a better way to hide fields on create, eg through an extension?

I'd recommend using permissions. If the user doesn't have permission to the field in the create permissions, it won't show up in the app either

**Comment by hibem** [comment]
set default  hidden , then set condition "Show it when id/date_created is not null"

**Reply by hibem** [reply]
In some cases, the ID , date_cteated will be invalid, so you can add the user_created or other fields that are only filled in after creation as a "and" condition
